src/EnergyPlus/WaterToAirHeatPumpSimple.cc

```diff
@@ -1134,122 +1086,100 @@ namespace WaterToAirHeatPumpSimple {
 
         // Set water and air inlet nodes
 
-        AirInletNode = state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).AirInletNodeNum;
-        WaterInletNode = state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).WaterInletNodeNum;
+        AirInletNode = simpleWatertoAirHP.AirInletNodeNum;
+        WaterInletNode = simpleWatertoAirHP.WaterInletNodeNum;
 
         if ((SensLoad != 0.0 || LatentLoad != 0.0) && (state.dataLoopNodes->Node(AirInletNode).MassFlowRate > 0.0)) {
-
-            // Model requires the values to be calculated at full design flow rate for air and then scaled to part load ratio.
-            // So always start the calculations by setting the air flow rate to design flow rate.
-
-            state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).WaterMassFlowRate =
-                state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).DesignWaterMassFlowRate;
-
-            state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).AirMassFlowRate =
-                state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).RatedAirVolFlowRate *
-                PsyRhoAirFnPbTdbW(state,
-                                  state.dataEnvrn->StdBaroPress,
-                                  state.dataLoopNodes->Node(AirInletNode).Temp,
-                                  state.dataLoopNodes->Node(AirInletNode).HumRat,
-                                  RoutineName);
-            // If air flow is less than 25% rated flow. Then set air flow to the 25% of rated conditions
-            if (state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).AirMassFlowRate <
-                0.25 * state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).RatedAirVolFlowRate *
-                    PsyRhoAirFnPbTdbW(state,
-                                      state.dataEnvrn->StdBaroPress,
-                                      state.dataLoopNodes->Node(AirInletNode).Temp,
-                                      state.dataLoopNodes->Node(AirInletNode).HumRat,
-                                      RoutineName)) {
-                state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).AirMassFlowRate =
-                    0.25 * state.dataWaterToAirHeatPumpSimple->SimpleWatertoAirHP(HPNum).RatedAirVolFlowRate *
-                    PsyRhoAirFnPbTdbW(state,
-                                      state.dataEnvrn->StdBaroPress,
-                                      state.dataLoopNodes->Node(AirInletNode).Temp,
-                                      state.dataLoopNodes->Node(AirInletNode).HumRat,
-                                      RoutineName);
+            simpleWatertoAirHP.WaterMassFlowRate = simpleWatertoAirHP.DesignWaterMassFlowRate;
+
+            simpleWatertoAirHP.AirMassFlowRate = state.dataLoopNodes->Node(AirInletNode).MassFlowRate;
```

Comment:
The model now reads the node flow rate and uses that information for calculations.